### PR TITLE
feat(Menu): new direction prop

### PR DESCRIPTION
**Pull request by sergivillar**
## Why?

We use this component in several places in the CMS, and we have some issues when the Menu appears on the right of the screen. We want to optionally allow select the align of the Menu items.

Screenshot 2021-08-27 at 11 16 49

### CODE REVIEW

File: src/__stories__/menu-story.tsx [hunk 1]
```diff
@@ -35,39 +45,42 @@ export const Default: StoryComponent = () => {
...
-                                    checked={valuesState.includes(option.value)}
-                                >
-                                    {option.text}
-                                </Checkbox>
-                            </Box>
-                        ))}
-                    </div>
-                )}
-            />
+            <div style={{display: 'flex', justifyContent: 'center'}}>
```
Comment (yceballost) [inline src/__stories__/menu-story.tsx:48]: can you add the menu as a snippet in Playroom? 🙏

**Review by yceballost — CHANGES_REQUESTED** [review]

File: src/__stories__/menu-story.tsx (hunk 1 above)
Comment (sergivillar) [reply src/__stories__/menu-story.tsx:48]: i'm not sure of what you want, is this link enough?

https://mistica-web.vercel.app/playroom#?code=N4Igxg9gJgpiBcIA6A7APAWRigrqgBPgO4CWUALgBYC8wALAAwMC%2BB%2BATtrOwCoCG7AOYxytABTAOMAGYAafBBQAFTgGdV8kqqy4A8gAds%2BZgEp81AHz4xbQmh4QcYSnwBGAGxj4YngLbZyACUZWk5pZgVlNVVaRRUYdQjVcgBPT1pJUgpKeHwARiZjZgtbQnw0AEkUdxIUL1V9PjAYWjyANmLSsvKKyBQAaRhXNx0cABkSQUpyfAB6EpRu7vsYAA9yAGYpQRx3AQtgLVGDIwB%2BfCQQAGF3CFUYS-xcy5OUS%2BY0WZ41zYWl8tmVRqdT%2BZU%2BDicLg8MFBJlYiykKG4o3EkjC8jAe3UADk%2BP4Mbd7sYzJZrKU0FASAA3KTSUIyCKYvg4vEtYBMln%2BToI7rAADaXTKwEF3XIP2eIAM5BIinyl1kIrKVL47hwMAlEH00sUeXliuYCp5S2FRv%2BYvWEqlMsWACY9aalsrVeqLiBNdqUHaQIb-oQDYKALoAOl8fH0YjE7utJKsNgddgAQhBVvhGlBKShBAANVodVN8dO1QQATVoAA4IgBrGApWJa61Bp1q7m%2BsFXSgwMCV1zJxXdFCsuseoPm8jw1vdRTtviZtliGP4E0TpYkaTWKOKRsqtXmajUV0blAbS5mJfLpb3cg8Ej%2BRzkCMLs-npaYu4wecAbj7voN%2BEYDBML94x-b9ukvABlcg%2BDFGwQEvAA1bcEnlBR603JsYEA0Cimw5xO2rKBaGEchIOg99LgQpDVBQvkAxMINakxHBYFUSM0JQLdnThb9QWXYBDxHH5xwnT523wntVl4pZPiTSTBRMbijU%2BSkqVheF5lQEBmCAA

Comment (yceballost) [reply src/__stories__/menu-story.tsx:48]: almost! add this code to playroom/snippets.tsx

File: src/__stories__/menu-story.tsx [hunk 2]
```diff
@@ -22,7 +28,11 @@ const options = [
     },
 ];
 
-export const Default: StoryComponent = () => {
+type MenuArgs = {
+    align: 'right' | 'left';
```
Comment (pladaria) [inline src/__stories__/menu-story.tsx:32]: to me `align` is a bad name (you aren't aligning), I'd use `openDirection` or just `direction`

File: src/menu.tsx [hunk 3]
```diff
@@ -64,37 +65,39 @@ const Menu: React.FC<MenuProps> = ({renderTarget, renderMenu, width}) => {
         }
 
         const MARGIN_THRESHOLD = 12;
-        const {top: selectTop, width, left, height} = targetRect;
-        const top = selectTop + height;
+        const {top: topTarget, width: widthTarget, left, right, height} = targetRect;
+        const top = topTarget + height;
         const spaceTaken = parseInt(window.getComputedStyle(menu).getPropertyValue('height')) ?? 0;
 
+        const leftAligned = align === 'left' ? left : right - (width ?? widthTarget);
```
Comment (pladaria) [inline src/menu.tsx:72]: same here, instead of `leftAligned` `shouldOpenToLeft` or something like that

File: src/__stories__/menu-story.tsx (hunk 2 above)
Comment (sergivillar) [reply src/__stories__/menu-story.tsx]: changed, but.... now that is called direction, `right` should be the default value? as the menu that open from the target to the right (as it is in the attached screenshot at the description of the PR) will be called right?

**Review by pladaria — APPROVED** [review]

File: src/menu.tsx [hunk 4]
```diff
@@ -45,9 +45,10 @@ export type MenuProps = {
     renderTarget: (props: TargetRenderProps) => React.ReactNode;
     renderMenu: (props: MenuRenderProps) => React.ReactNode;
     children?: void;
+    direction?: 'left' | 'right';
```
Comment (pladaria) [inline src/menu.tsx:48]: component API has changed, you should execute `yarn build` and commit updated type definitions

**Review by pladaria — APPROVED** [review]

**Comment by atabel** [comment]
shouldn't this be automatic?

**Review by yceballost — APPROVED** [review]

File: src/__stories__/menu-story.tsx (hunk 2 above)
Comment (atabel) [reply src/__stories__/menu-story.tsx]: in Toltip we call this `position`

**Comment by sergivillar** [comment]
> shouldn't this be automatic?

It could be, in our us case we will set this manually always, and I dont want to add more complexity to this component, but, if you want, I could try yo make this automatic, WDYT?

**Review by atabel — APPROVED** [review]

File: src/__stories__/menu-story.tsx (hunk 2 above)
Comment (sergivillar) [reply src/__stories__/menu-story.tsx]: changed

**Pull request merged**